game/src/main/scala/io/aigar/game/EntityContainer.scala

```diff
@@ -0,0 +1,22 @@
+package io.aigar.game
+
+import com.github.jpbetz.subspace.Vector2
+
+
+trait EntityContainer {
+  def respawn(grid: Grid, players: List[Player], respawnRetryAttempts: Int): Option[Vector2] = {
+    var newPosition = grid.randomPosition
+
+    1 to respawnRetryAttempts foreach { _ =>
+      for (player <- players) {
+        for (cell <- player.cells) {
+          if (cell.contains(newPosition))
+            newPosition = grid.randomPosition
+          else
+            return Some(newPosition)
```

Comment:
Riiiight. My bad. I misread the code. It still doesn't work though: if the first cell does not contain `newPosition`, it will return right away, **even if** the second cell (or the cell of another player) contains the position.

Also, I don't think we should do `newPosition = grid.randomPosition` *while* iterating through the cells, or else we could do the following: `no collision, no collision, no collision, ... OOPS collision respawn (RIGHT on the first cell), no collision... All good, return this one` while we just created a new collision as we were iterating.

I think we need something along the lines of:
```scala
    1 to respawnRetryAttempts foreach { _ =>
        collides = false
        for players ...
            for cells ....
                if (cell.contains(newPosition)) collides = true
        if (!collides) return newPosition
        newPosition = grid.randomPosition
    }
    return newPosition
```